internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
Private env allows more flexibility than mock though. For eg. an empty test env with no workflows or activities. I don't see any downside to keeping this around and it provides added flexibility. In the newTestWorkflowEnv we can always set private env to the global getHostEnv so thats always the case by default. If we choose to override in tests now or later, we still can.